opm/material/components/H2O.hpp

```diff
@@ -981,13 +981,24 @@ class H2O : public Component<Scalar, H2O<Scalar> >
                                    const Evaluation& temperature,
                                    const Evaluation& pressure)
     {
-        auto tostring = [](const auto& val) -> std::string
+        auto cast = [](const auto d)
         {
-            if constexpr (DenseAd::is_evaluation<Evaluation>::value)
-                return std::to_string(getValue(val.value()));
+#if HAVE_QUAD
+            if constexpr (std::is_same_v<decltype(d), const quad>)
+                return static_cast<double>(d);
```

Comment:
Probably making a fool out of myself with this question because of my ignorance:
Why does  a simple `static_cast<double>(d)` not work for all types? Isn't d always floating point?

Reply:
It should work but it might change output with float, so i was being defensive..